raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java

```diff
@@ -92,4 +100,22 @@ public static Path createTempFile(Path logDir, OffsetAndEpoch snapshotId) throws
 
         return Optional.of(new SnapshotPath(path, new OffsetAndEpoch(endOffset, epoch), partial));
     }
+
+    /**
+     * Delete this snapshot from the filesystem.
+     */
+    public static CompletableFuture<Boolean> deleteSnapshotIfExists(Path logDir, OffsetAndEpoch snapshotId) throws IOException {
+        Path path = snapshotPath(logDir, snapshotId);
+        Path destination = Snapshots.deleteRename(path, snapshotId);
+        Utils.atomicMoveWithFallback(path, destination);
+
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return Files.deleteIfExists(destination);
+            } catch (IOException e) {
+                throw new RuntimeException("Error deleting snapshot file " + destination + ":" + e.getMessage());
+            }
+        });
```

Comment:
This uses the default Executor in Java. I think we should instead use the same async scheduler that is used for the `KafkaMetadataLog`.

Reply:
I found that KafkaMetadataLog doesn't use a scheduler currently, should we add a scheduler to it?

Reply:
That's correct, it indirectly uses the scheduler in `Log.scala` when deleting segment for example.

Yes. We can pass the same scheduler that we pass to `Log.scala` to this type when it is constructed.

Reply:
Added schedule to `KafkaMetadataLog`, PTAL. Do you think we should add a `MetadataLogSnapshot` class instead of using `OffsetAndEpoch` to represent a snapshot?